File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.cpp
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> m1;
        unordered_set<int> m2;
        vector<int> ans;
        for(int i=0;i<nums1.size();i++)
        {
            m1.insert(nums1[i]);
        }
        for(int i=0;i<nums2.size();i++)
        {
            if(m1.count(nums2[i])==1)
                m2.insert(nums2[i]);
            else
                continue;
        }
        for(auto i:m2)
        {
            ans.emplace_back(i);
        }
        return ans;
    }
};

File: 0349-intersection-of-two-arrays/0349-intersection-of-two-arrays.java
class Solution {
    public int[] intersection(int[] nums1, int[] nums2) {
        HashSet<Integer> h1 = new HashSet<>();
        HashSet<Integer> h2 = new HashSet<>();
        int n = nums1.length;
        int m = nums2.length;
        for(int i=0;i<n;i++)
        {
            h1.add(nums1[i]);
        }
        for(int i=0;i<m;i++)
        {
            if(h1.contains(nums2[i]))
            {
                h2.add(nums2[i]);
            }
        }
        int arr[] = new int[h2.size()];
        int j=0;
        for(int ele:h2)
        arr[j++]=ele;
        return arr;
    }
}